Support disable installer

mengqiy opened the pull request:
This supersedes #158.

droot commented:
@mengqiy Took a quick look, looks good. Will take another detailed look in the afternoon.

mengqiy commented:
PTAL

droot commented on example/main.go:
```diff
@@ -39,18 +39,24 @@ import (
+	var enableInstaller bool
+	flag.BoolVar(&enableInstaller, "enable-installer", false,
+		"enable the installer in the webhook server, so it will install webhook related resources during bootstrapping")
+
```
shall we rename the flag to reflect the usage ? for ex. `--install-webhook-config` ?

droot commented on pkg/webhook/server.go:
```diff
@@ -54,10 +54,9 @@ type ServerOptions struct {
-	Dryrun bool
+	// EnableInstaller controls if the server will automatically create webhook related objects
+	// during bootstrapping. e.g. webhookConfiguration, service and secret.
+	EnableInstaller bool
```
May be rename the flag ? ex. `InstallWebhookConfig` ?

droot commented on pkg/webhook/server.go:
```diff
@@ -187,13 +187,17 @@ func (s *Server) Handle(pattern string, handler http.Handler) {
-	if err != nil || s.Dryrun {
-		return err
+	if s.EnableInstaller {
+		log.Info("webhook installer is enabled")
```
We can improve the log --> installing webhook configuration in cluster ? or something for user-friendly.

mengqiy commented:
Addressed comments.
Pushed one more commit to address an issue due to https://github.com/kubernetes/apimachinery/blob/master/pkg/runtime/register.go#L26-L29

droot commented on pkg/webhook/internal/cert/writer/secret_test.go:
```diff
@@ -78,6 +79,10 @@ var _ = Describe("secretCertWriter", func() {
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "v1",
+					Kind:       "Secret",
+				},
```
As per my understanding, as long as the client has correct scheme (in this case scheme for core-types which is set by default), we don't need to set typemeta. Has anything changed ?
